### ISSUE TITLE
Remove Linaria mocks from Jest global setup

### ISSUE THREAD

**Issue by kurtdoherty** [reapit/elements#158]
### Context

TL;DR The global mocks we currently setup in `scripts/jest/jest-setup.js` for Linaria's `css` and `styled` utilities interfere with React Testing Library's DOM queries because they result in non-standard tag names being used (e.g. `<mock-styled.textarea>`).

This blocks our ability to write effective unit tests for components, effectively restricting us to the sole use of DOM snapshots.

The mocks for Linaria were required with previous versions of Linaria, but they are no longer necessary. See #152 for more details.

### Work to be done

I see two main options here:

1. Like #157, we remove the mocks and do a big bang update for all the snapshots; or,
2. We defer this work in favour of switching to Vitest (at which point, the Jest mocks for Linaria won't be relevant, and the snapshot will be replaced with Vitest-written versions). This change is proposed in #159.

At this stage, I'm in favour of (2), but that's based on the assumption that we'll actually agree to switch from Jest to Vitest.
